ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml

```diff
@@ -381,11 +383,104 @@ StatusScrollView {
+
+            useLetterIdenticons: !wholeCommunitySelected || !inDropdown.communityImage
+
+            property bool wholeCommunitySelected: false
+
+            ListModel {
+                id: inModelCommunity
+
+                readonly property string colorWorkaround: inDropdown.communityData.color
+
+                Component.onCompleted: {
+                    append({
+                        imageSource: inDropdown.communityData.image,
+                        text: inDropdown.communityData.name,
+                        operator: OperatorsUtils.Operators.None
+                    })
+
+                    setProperty(0, "color", colorWorkaround)
+                }
+            }
+
+            ListModel {
+                id: inModelChannels
+            }
+
+            InDropdown {
+                id: inDropdown
+
+                model: root.rootStore.chatCommunitySectionModule.model
+
+                readonly property var communityData: rootStore.mainModuleInst.activeSection
+
+                communityName: communityData.name
+                communityImage: communityData.image
+                communityColor: communityData.color
+
+                onChannelsSelected: {
+                    inModelChannels.clear()
+                    inSelector.itemsModel = 0
+                    inSelector.wholeCommunitySelected = false
+
+                    channels.map(channel => {
+                        inModelChannels.append({
+                            itemId: channel.itemId,
+                            text: "#" + channel.name,
+                            emoji: channel.emoji,
+                            color: channel.color,
+                            operator: OperatorsUtils.Operators.None
+                        })
+                    })
+
+                    inSelector.itemsModel = inModelChannels
+                    close()
+                }
+
+                onCommunitySelected: {
+                    inModelChannels.clear()
+                    inSelector.wholeCommunitySelected = true
+                    inSelector.itemsModel = inModelCommunity
+                    close()
+                }
+            }
+
+            function openInDropdown(parent, x, y) {
+                inDropdown.parent = parent
+                inDropdown.x = x
+                inDropdown.y = y
+
+                const selectedChannels = []
+
+                if (!inSelector.wholeCommunitySelected)
+                    for (let i = 0; i < inModelChannels.count; i++)
+                        selectedChannels.push(inModelChannels.get(i).itemId)
+
+                inDropdown.setSelectedChannels(selectedChannels)
+                inDropdown.open()
+            }
+
+            addButton.onClicked: {
+                inDropdown.acceptMode = InDropdown.AcceptMode.Add
+                openInDropdown(inSelector.addButton,
+                               inSelector.addButton.width + 4, 0)
```

Comment:
Maybe we can use a readonly property for this magic number `4` that is being repeated along this component and it exactly defines the same / used for the same propose. WDYT?